spot_driver_cpp/test/src/images/test_spot_image_publisher_node.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <gmock/gmock.h>
+
+#include <spot_driver_cpp/api/image_client_api.hpp>
+#include <spot_driver_cpp/api/spot_api.hpp>
+#include <spot_driver_cpp/images/spot_image_publisher_node.hpp>
+
+#include <spot_driver_cpp/mock/mock_image_client_api.hpp>
+#include <spot_driver_cpp/mock/mock_logger_interface.hpp>
+#include <spot_driver_cpp/mock/mock_spot_api.hpp>
+
+#include <rclcpp/node.hpp>
+#include <rclcpp/node_options.hpp>
+
+#include <exception>
+#include <memory>
+#include <optional>
+#include <tl_expected/expected.hpp>
+
+using ::testing::_;
+using ::testing::InSequence;
+using ::testing::Return;
+
+namespace spot_ros2::images::test {
+
+class FakeParameterInterface : public ParameterInterfaceBase {
+ public:
+  std::string getAddress() const override { return address; }
+
+  std::string getUsername() const override { return username; }
+
+  std::string getPassword() const override { return password; }
+
+  double getRGBImageQuality() const override { return rgb_image_quality; }
+
+  bool getHasRGBCameras() const override { return has_rgb_cameras; }
+
+  bool getPublishRGBImages() const override { return publish_rgb_images; }
+
+  bool getPublishDepthImages() const override { return publish_depth_images; }
+
+  bool getPublishDepthRegisteredImages() const override { return publish_depth_registered_images; }
+
+  std::string getSpotName() const override { return spot_name; }
+
+  std::string address;
+  std::string username;
+  std::string password;
+
+  double rgb_image_quality = kDefaultRGBImageQuality;
+  bool has_rgb_cameras = kDefaultHasRGBCameras;
+  bool publish_rgb_images = kDefaultPublishRGBImages;
+  bool publish_depth_images = kDefaultPublishDepthImages;
+  bool publish_depth_registered_images = kDefaultPublishDepthRegisteredImages;
+  std::string spot_name;
+};
+
+class SpotImagePubNodeTestFixture : public ::testing::Test {
+ public:
+  void SetUp() override {
+    rclcpp::init(0, nullptr);
+    node = std::make_shared<rclcpp::Node>("test_image_publisher_node");
```

Comment:
The presence of these within the test is a sign that the implementation of these classes needs to be tweaked a little bit: we should be able to unit test SpotImagePublisherNode without having an actual instance of `rclcpp::Node` or initializing the ROS 2 context.